eng/pipelines/templates/steps/compilation.check.yml

```diff
@@ -26,14 +26,14 @@ steps:
         exit 1
       }
       foreach ($retry in 0..${{ parameters.RetryLimit }}) {
-        Write-Host "mvn --no-transfer-progress -f $pomf compile"
-        mvn --no-transfer-progress -f $pomf compile
-        if (!$LASTEXITCODE -or ($retry -ge ${{ parameters.RetryLimit }})) { break }
+        Write-Host "mvn --no-transfer-progress --update-snapshots -f $pomf clean compile"
+        mvn --no-transfer-progress --update-snapshots -f $pomf clean compile
+        if (!$LASTEXITCODE) { exit 0 }
         Write-Host "Install failed, retrying in 1 minute..."
         Start-Sleep -s 60
       }
 
-      exit $LASTEXITCODE
```

Comment:
Is there any harm in preserving the maven exit code?

Reply:
I don't think there is any harm in preserving it, my thoughts on this change were to be more explicit. As in, if we reached here we know we failed without a doubt, so let's not have any potential `LASTEXITCODE` mutations accidentally return a passed state